src/readers/morphologyHDF5.cpp

```diff
@@ -70,193 +87,154 @@ Property::Properties load(const HighFive::Group& group) {
 
 Property::Properties MorphologyHDF5::load() {
     _readMetadata(_uri);
-    int firstSectionOffset = _readSections();
-    _readPoints(firstSectionOffset);
-    _readPerimeters(firstSectionOffset);
-    _readMitochondria();
-    _readEndoplasmicReticulum();
-    std::get<0>(_properties._cellLevel._version) = "h5";
-
-    return _properties;
-}
 
-MorphologyHDF5::MorphologyHDF5(const HighFive::Group& group)
-    : _group(group)
-    , _uri("HDF5 Group") {}
+    int firstSectionOffset = _readSections();
 
+    _readPoints(firstSectionOffset);
 
-void MorphologyHDF5::_resolveV1() {
-    HighFive::SilenceHDF5 silence;
-    _points.reset(new HighFive::DataSet(_group.getDataSet(_d_points)));
-    auto dataspace = _points->getSpace();
-    _pointsDims = dataspace.getDimensions();
+    if (_properties._cellLevel.minorVersion() >= 1) {
+        _readPerimeters(firstSectionOffset);
 
-    if (_pointsDims.size() != 2 || _pointsDims[1] != _pointColumns) {
-        throw morphio::RawDataError("Opening morphology '" + _uri +
-                                    "': bad number of dimensions in 'points' dataspace.");
+        if (_properties._cellLevel.minorVersion() >= 2) {
+            _readMitochondria();
+            _readEndoplasmicReticulum();
+        }
```

Comment:
Does it means if we have a version 1.2 we will always try to read the mito/endo data even if we are opening a neuron ?

Reply:
my bad missed the :
```
if (!_group.exist(_g_mitochondria)) {
        return;
    }
```
and
```
if (!_group.exist(_g_endoplasmic_reticulum)) {
        return;
    }
```